### PR TITLE
Added the Git integration to segments, and tweaks for the GitHub integration

### PR DESCRIPTION
**Pull request by joanreyero**
# Changes proposed ✍️

### <samp>🤖 Generated by Copilot at 13424bc</samp>

This pull request adds segment support to the backend services and repositories, allowing multiple git integrations per platform and segment. It also changes the order of the streams created by the `GithubIntegrationService` and links the activities to their conversations. It also adds some console logs for debugging purposes.
​

### <samp>🤖 Generated by Copilot at 13424bc</samp>

> _We're fetching all the integrations by platform and segment_
> _We're linking the activities to their conversations_
> _We're logging the segments for debugging and testing_
> _Heave away, me hearties, heave away on the count of three_

### <samp>🤖 Generated by Copilot at 13424bc</samp>

*  Add a method to fetch all integrations by platform and tenant ([link](https://github.com/CrowdDotDev/crowd.dev/pull/994/files?diff=unified&w=0#diff-5b1c50661ca3f395a4ec888993b46eadd35b277a777e88e0e81e1f88461124a3R158-R176))
*  Modify the order of the streams created by the GitHub integration service ([link](https://github.com/CrowdDotDev/crowd.dev/pull/994/files?diff=unified&w=0#diff-93f2daaa968f5bec5c40cd31fc27ac6422ea3dcbdabecf3ef80f3c16802d1c30L236-R236))
*  Add a method to fetch all git remotes for different segments and use it in the gitGetRemotes method ([link](https://github.com/CrowdDotDev/crowd.dev/pull/994/files?diff=unified&w=0#diff-80e1c7e63674653fe2cf5bcbe4a2212499d72014f328ad2e6b617146f0a256fdR94-R97), [link](https://github.com/CrowdDotDev/crowd.dev/pull/994/files?diff=unified&w=0#diff-80e1c7e63674653fe2cf5bcbe4a2212499d72014f328ad2e6b617146f0a256fdL617-R629))
*  Add a call to link the activity to its parent conversation ([link](https://github.com/CrowdDotDev/crowd.dev/pull/994/files?diff=unified&w=0#diff-42eeafe29b34fac65b490c471e565ea8c89bcc468bf0af67a09ecf20a496cf1cR99))
*  Import and instantiate the segment repository in the db operations handler ([link](https://github.com/CrowdDotDev/crowd.dev/pull/994/files?diff=unified&w=0#diff-39fce494148150c242c2649b6a7ed2867ec4cd63607581895ad5faa47e518f4eR5), [link](https://github.com/CrowdDotDev/crowd.dev/pull/994/files?diff=unified&w=0#diff-39fce494148150c242c2649b6a7ed2867ec4cd63607581895ad5faa47e518f4eR26-R29))
*  Add console logs to debug the segment id and the segment options in the member and sequelize repositories (`memberRepository.ts`, `sequelizeRepository.ts`, [link](https://github.com/CrowdDotDev/crowd.dev/pull/994/files?diff=unified&w=0#diff-104433e3d1e0684596dd40f3f2c65b83799a74181ad8e7c74fc52560800a46fbL54-R55), [link](https://github.com/CrowdDotDev/crowd.dev/pull/994/files?diff=unified&w=0#diff-9db36b6ff29d8472a654666df43dcbe269272d189dac126447c3ffa8751f7d87L69-R69))

## Checklist ✅
- [ ] Label appropriately with `Feature`, `Improvement`, or `Bug`.
- [ ] Add screehshots to the PR description for relevant FE changes
- [ ] New backend functionality has been unit-tested.
- [ ] API documentation has been updated (if necessary) (see [docs on API documentation](https://docs.crowd.dev/docs/updating-api-documentation)).
- [ ] [Quality standards](https://github.com/CrowdDotDev/crowd-github-test-public/blob/main/CONTRIBUTING.md#quality-standards) are met.

### CODE REVIEW

File: backend/src/serverless/dbOperations/handler.ts [hunk 1]
```diff
@@ -22,6 +23,8 @@ export async function consumer(event) {
   }
 
   const context = await getUserContext(tenantId)
+  const segmentRepository = new SegmentRepository(context)
+  context.currentSegments = [await segmentRepository.findById(event.segments[0])]
```
Comment (sausage-todd) [inline backend/src/serverless/dbOperations/handler.ts:27]: Shouldn't we use all segments from `event.segments` ?
Where is this `event.segments` coming from? From python worker?

**Review by sausage-todd — APPROVED** [review]
Looks good apart from that one question

File: backend/src/serverless/dbOperations/handler.ts (hunk 1 above)
Comment (joanreyero) [reply backend/src/serverless/dbOperations/handler.ts:27]: From the Git integration. Since it's an integration, it will only ever be 1 segment.

Comment (sausage-todd) [reply backend/src/serverless/dbOperations/handler.ts:27]: Does it make sense to send only one segment in the first place?

**Pull request merged**